### PR TITLE
[homeconnect] Delay update of state options when not accessible

### LINKED ISSUE #10705: [homeconnect] Washing program with missing options
Sometimes the washing program options are returned empty by the API when called by the binding:

image

When I show the details:

image

The consequence is that options are not built for the spin speed and the washing temperature channels. And as the result is cached by the binding, this will remain as it is until a restart of the binding.

But then if I run manually the same request from the UI page, it works as expected:

image

### PR DESCRIPTION
**Pull request by lolodomo**
Fix #10705

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
@bruestel : please note I moved the program options cache in the thing handler. I think 2 devices could have the same program with different options. For example, you could imagine two washers with a coton program, one will offer 1400 RPM spin speed and the other will not.

**Comment by lolodomo** [comment]
The build failed in tradfri binding integration tests (due to new version of jmdns).
So no link with my PR.

File: bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/handler/AbstractHomeConnectThingHandler.java [hunk 1]
```diff
@@ -855,15 +859,41 @@ protected EventHandler defaultAmbientLightCustomColorStateEventHandler() {
...
     }
 
+    protected EventHandler updateRemoteControlActiveAndProgramOptionsStateEventHandler() {
+        return event -> {
+            defaultBooleanEventHandler(CHANNEL_REMOTE_CONTROL_ACTIVE_STATE).handle(event);
+
+            // update available program options if update was previously delayed and remote control is enabled
+            try {
+                String programKey = programOptionsDelayedUpdate;
+                if (programKey != null && "true".equals(event.getValue())) {
```
Comment (bruestel) [inline bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/handler/AbstractHomeConnectThingHandler.java:869]: Could you please use `parseBoolean` (https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Boolean.html#parseBoolean(java.lang.String))

**Review by bruestel — CHANGES_REQUESTED** [review]
Looks good,

File: bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/handler/AbstractHomeConnectThingHandler.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/handler/AbstractHomeConnectThingHandler.java:869]: Done

**Review by bruestel — APPROVED** [review]

**Review by fwolter — APPROVED** [review]
LGTM

**Pull request merged**